scheduler/scheduler.go

```diff
@@ -526,29 +629,54 @@ func (driver *MesosSchedulerDriver) Start() (mesos.Status, error) {
 	driver.setStatus(mesos.Status_DRIVER_RUNNING)
 	driver.setStopped(false)
 
-	log.Infoln("Mesos scheduler driver started with PID=", driver.self.String())
+	log.Infof("Mesos scheduler driver started with PID=%v", driver.self)
 
+	// start the detector AFTER we have a self pid from the messenger, otherwise things get ugly.
```

Comment:
that comment isn't in the right place, it should be right before the Detect() call. it also could use some clarification: "setup the detect callback AFTER we have a self pid from the messenger, otherwise..."